pkg/hubble/monitor/filter_test.go

```diff
@@ -0,0 +1,497 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package monitor
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/assert"
+
+	observerTypes "github.com/cilium/cilium/pkg/hubble/observer/types"
+	"github.com/cilium/cilium/pkg/hubble/parser/errors"
+	monitorAPI "github.com/cilium/cilium/pkg/monitor/api"
+)
+
+func TestNewMonitorFilter(t *testing.T) {
+	logger, hook := test.NewNullLogger()
+
+	tests := []struct {
+		name        string
+		filters     []string
+		expectedErr error
+		expectedMF  *monitorFilter
+	}{
+		{
+			name:        "unknown filter",
+			filters:     []string{"unknown"},
+			expectedErr: fmt.Errorf("unknown monitor event type: unknown"),
+			expectedMF:  nil,
+		},
+		{
+			name: "valid filters",
+			filters: []string{
+				monitorAPI.MessageTypeNameDrop,
+				monitorAPI.MessageTypeNameDebug,
+				monitorAPI.MessageTypeNameCapture,
+				monitorAPI.MessageTypeNameTrace,
+				monitorAPI.MessageTypeNameL7,
+				monitorAPI.MessageTypeNameAgent,
+				monitorAPI.MessageTypeNamePolicyVerdict,
+				monitorAPI.MessageTypeNameRecCapture,
+				monitorAPI.MessageTypeNameTraceSock,
+			},
+			expectedErr: nil,
+			expectedMF: &monitorFilter{
+				logger: logger,
+
+				drop:          true,
+				debug:         true,
+				capture:       true,
+				trace:         true,
+				l7:            true,
+				agent:         true,
+				policyVerdict: true,
+				recCapture:    true,
+				traceSock:     true,
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			mf, err := NewMonitorFilter(logger, tc.filters)
+			assert.Equal(t, tc.expectedErr, err)
+			assert.Equal(t, tc.expectedMF, mf)
+
+			if tc.expectedMF != nil {
+				assert.Equal(t, 1, len(hook.Entries))
+				assert.Equal(t, logrus.InfoLevel, hook.LastEntry().Level)
+				assert.Equal(t, "Configured Hubble with monitor event filters", hook.LastEntry().Message)
+				assert.Equal(t, tc.filters, hook.LastEntry().Data["filters"])
+			}
+
+			hook.Reset()
+		})
+	}
+}
+
+type testEvent struct {
+	event       *observerTypes.MonitorEvent
+	allowed     bool
```

Comment:
to be consistent with https://github.com/cilium/cilium/blob/main/pkg/hubble/observer/local_observer.go#L119 please rename `allowed` to `stop` or `stopPropagation`